Adding basic schema validation to damsrepo (DHH-602)

escowles opened the pull request:
- Performing basic schema validation (requiring classes and predicates to be present in a list of valid URIs)
- Fixing invalid sample data.
- Validating input for both bulk loading and REST API updates.

escowles committed:
Adding basic schema validation to damsrepo (DHH-602)

mcritchlow commented on src/java/edu/ucsd/library/dams/triple/TripleStoreUtil.java:
```diff
@@ -320,6 +335,49 @@ private static void loadRDF( InputStream in, boolean deleteFirst,
+
+		return null;
+	}
+	public static Set<String> validateModel( Model model, Set<String> validClasses,
```
If our list of validations continues to grow beyond core class/predicate matching, I wonder if this belongs in its own class, with each of the validation checks being in a separate method. I know we'll eventually port this to Ruby, so I'm not strongly opinionated on this.

escowles replied:
That could be a good idea, since I wouldn't be surprised if we do wind up implementing more validation operations in the short/medium term.  Though I was hoping to keep as much of that as possible in Ruby to make it as easy as possible to migrate.

mcritchlow replied:
@escowles Would you prefer this merged then for now, and we revisit if/when we get additional validation requirements?

escowles committed:
Moving validation to its own class, fixing sample records (DHH-602)

escowles commented:
Having the validation code in its own class is a better design -- I've moved validateModel to a new class called Validator and fixed an invalid type in the sample data (DHH-602).
